.circleci/config.yml

```diff
@@ -0,0 +1,71 @@
+# Golang CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-go/ for more details
+version: 2
+jobs:
+  build:
+    working_directory: /go/src/github.com/ecadlabs/tezos_exporter
+    docker:
+      - image: circleci/golang:1.11
+
+    steps:
+    - checkout
+    - run: go build
+
+  docker_hub_branch:
+    docker:
+      - image: circleci/golang:1.11
+    working_directory: /go/src/github.com/ecadlabs/tezos_exporter
+
+    steps:
+    - checkout
+    - setup_remote_docker
+    - attach_workspace:
+        at: .
+    - run: docker build -t ecadlabs/tezos_exporter:$CIRCLE_BRANCH -t ecadlabs/tezos_exporter:${CIRCLE_SHA1:0:8} .
+    - run: docker images
+    - run: docker login -u $DOCKER_USER -p $DOCKER_PASS
+    - run: docker push ecadlabs/tezos_exporter:$CIRCLE_BRANCH
+    - run: docker push ecadlabs/tezos_exporter:${CIRCLE_SHA1:0:8}
+
+  docker_hub_release_tags:
+    docker:
+      - image: circleci/golang:1.11
+    working_directory: /go/src/github.com/ecadlabs/tezos_exporter
+
+    steps:
+    - checkout
+    - setup_remote_docker
+    - attach_workspace:
+        at: .
+    - run: docker build -t ecadlabs/tezos_exporter:$CIRCLE_TAG -t ecadlabs/tezos_exporter:latest .
+    - run: docker images
+    - run: docker login -u $DOCKER_USER -p $DOCKER_PASS
+    - run: docker push ecadlabs/tezos_exporter:$CIRCLE_TAG
+    - run: docker push ecadlabs/tezos_exporter:latest
+
+
+workflows:
+  version: 2
+  tezos_exporter:
+    jobs:
+    - build:
+        filters:
+          tags:
+            only: /.*/
+    - docker_hub_branch:
+        requires:
+          - build
+        filters:
+          branches:
+            only: /.*/
```

Comment:
Do I understand correctly that this will build Docker images for all branches, including ephemeral PR ones? Is that desired?

Reply:
Yes, that's too permissive, left over from debugging CI. I'll tighten it up.